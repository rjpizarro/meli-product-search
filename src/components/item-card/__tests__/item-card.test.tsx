import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ItemCard from '../index'

jest.mock('../../../components/price-label', () => () => <div data-testid="price-label" />)
jest.mock('../../../components/icons', () => ({
    ShippingIcon: () => <div data-testid="shipping-icon" />
}))

const item = {
    id: "MOCK-ID-1",
    title: "Test item 1",
    condition: "new",
    description: "Test description 1",
    picture: "test picture 1",
    sold_quantity: 10,
    price: {
        amount: 100,
        decimals: 0
    }
}

describe('ItemCard', () => {
    it('should render correctly ', () => {
        const tree = renderer
            .create(<ItemCard {...item} />)
            .toJSON()
        const { queryByTestId }  = render(<ItemCard {...item} />);

        expect(tree).toMatchSnapshot()
        expect(queryByTestId('shipping-icon')).not.toBeInTheDocument()
    })

    it('should render a icon when free_shipping prop is true', () => {
        const { queryByTestId }  = render(<ItemCard {...item} free_shipping={true} />);
        const tree = renderer
            .create(<ItemCard {...item} free_shipping={true} />)
            .toJSON()

        expect(tree).toMatchSnapshot()
        expect(queryByTestId('shipping-icon')).toBeInTheDocument()
    })
})

