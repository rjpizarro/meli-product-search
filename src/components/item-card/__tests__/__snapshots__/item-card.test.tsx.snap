// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard should render a icon when free_shipping prop is true 1`] = `
<div
  className="item-card"
>
  <div
    className="item-card__picture-container"
  >
    <img
      alt="Test item 1"
      className="item-card__picture"
      src="test picture 1"
    />
  </div>
  <div
    className="item-card__details-container"
  >
    <div
      className="item-card__price-details"
    >
      <div
        data-testid="price-label"
      />
      <span
        className="item-card__shipping-icon"
      >
        <div
          data-testid="shipping-icon"
        />
      </span>
    </div>
    <h2
      className="item-card__title"
    >
      Test item 1
    </h2>
  </div>
  <div
    className="item-card__location-container"
  >
    <span
      className="item-card__location"
    />
  </div>
</div>
`;

exports[`ItemCard should render correctly  1`] = `
<div
  className="item-card"
>
  <div
    className="item-card__picture-container"
  >
    <img
      alt="Test item 1"
      className="item-card__picture"
      src="test picture 1"
    />
  </div>
  <div
    className="item-card__details-container"
  >
    <div
      className="item-card__price-details"
    >
      <div
        data-testid="price-label"
      />
    </div>
    <h2
      className="item-card__title"
    >
      Test item 1
    </h2>
  </div>
  <div
    className="item-card__location-container"
  >
    <span
      className="item-card__location"
    />
  </div>
</div>
`;
